h2/src/main/org/h2/result/LocalResult.java

```diff
@@ -420,12 +436,30 @@ private void trimExternal(int offset, int limit) {
         while (--offset >= 0) {
             temp.next();
         }
+        Value[] row = null;
         while (--limit >= 0) {
-            rows.add(temp.next());
+            row = temp.next();
+            rows.add(row);
             if (rows.size() > maxMemoryRows) {
                 addRowsToDisk();
             }
         }
+        if (withTies && sort != null && row != null) {
+            int[] indexes = sort.getQueryColumnIndexes();
+            Value[] expected = row;
+            loop: while ((row = temp.next()) != null) {
+                for (int idx : indexes) {
+                    if (!expected[idx].equals(row[idx])) {
+                        break loop;
```

Comment:
I think continue is easier to read to "break loop", but that' just a taste preference